### PR TITLE
add last_modified functionality

### PR DESCRIPTION
**Pull request by MDavidson17**
Fixes: #301  

### Change Description:
New Column added to buildings.building_outlines (last_modified) that is updated with new functions when building_outlines are edited (attribute or geometry changes)
The last_modified field of the buildings_lds tables are now updated to be set as the most recent of begin_lifespan, end_lifespan and last_modified. 

### Notes for Testing:
- Test data updated
- db and plugin tests add

#### Source Code Documentation Tasks:
- [ ] README updated (where applicable)
- [ ] CHANGELOG (Unreleased section) updated
- [ ] Docstrings / comments included to help explain code

#### User Documentation Tasks:
- [ ] Confluence user guide updated (where applicable)

#### Testing Tasks:
- [ ] Added tests that fail without this change
- [ ] All tests are passing in development environment
- [ ] Reviewers assigned
- [ ] Linked to main issue for ZenHub board

### CODE REVIEW

File: db/sql/revert/buildings/functions/building_outlines.sql [hunk 1]
```diff
@@ -1,23 +1,334 @@
--- Revert nz-buildings:buildings/functions/building_outlines from pg
+-- Revert nz-buildings:buildings/functions/building_outlines to v1.0.4
 
 BEGIN;
 
-DROP FUNCTION buildings.building_outlines_insert(integer, integer, integer, integer, integer, integer, integer, geometry);
+--------------------------------------------
```
Comment (YingtingChen) [inline db/sql/revert/buildings/functions/building_outlines.sql]: I guess this should be `DROP FUNCTION`?

File: db/sql/revert/buildings_lds/functions/populate_buildings_lds.sql [hunk 2]
```diff
@@ -1,13 +1,232 @@
...
 
 BEGIN;
 
-DROP FUNCTION buildings_lds.nz_building_outlines_insert();
+------------------------------------------------------------------------------
+-- Populate buildings_lds schema in preparation for loading data to LDS
+------------------------------------------------------------------------------
 
-DROP FUNCTION buildings_lds.nz_building_outlines_all_sources_insert();
+CREATE OR REPLACE FUNCTION buildings_lds.nz_building_outlines_insert()
```
Comment (YingtingChen) [inline db/sql/revert/buildings_lds/functions/populate_buildings_lds.sql]: Same as the other one, it should be `DROP FUNCTION`

File: db/sql/revert/buildings/functions/building_outlines.sql (hunk 1 above)
Comment (dwsilk) [reply db/sql/revert/buildings/functions/building_outlines.sql]: No, this code sets it back to `v1.4.0` of the function, not back to nothing.

File: db/sql/revert/buildings/functions/building_outlines.sql [hunk 3]
```diff
@@ -1,23 +1,334 @@
--- Revert nz-buildings:buildings/functions/building_outlines from pg
+-- Revert nz-buildings:buildings/functions/building_outlines to v1.0.4
```
Comment (dwsilk) [inline db/sql/revert/buildings/functions/building_outlines.sql]: This should be `v1.4.0-dev1`? How did it end up being `v1.0.4`?

File: db/sql/revert/buildings/functions/building_outlines.sql (hunk 1 above)
Comment (YingtingChen) [reply db/sql/revert/buildings/functions/building_outlines.sql]: I see what you meant this morning. The `sqitch rework` will copy the deploy file of the older version to revert :grimacing:

**Pull request merged**